overlord/ifacestate/ifacemgr.go

```diff
@@ -285,7 +303,21 @@ func (m *InterfaceManager) doConnect(task *state.Task, _ *tomb.Tomb) error {
 	if err != nil {
 		return err
 	}
-	return m.repo.Connect(plugRef.Snap, plugRef.Name, slotRef.Snap, slotRef.Name)
+
+	conns, err := getConns(task)
+	if err != nil {
+		return err
+	}
+
+	if err := m.repo.Connect(plugRef.Snap, plugRef.Name, slotRef.Snap, slotRef.Name); err != nil {
```

Comment:
Let's please break this line down so the code is more readable (err != nil on its own).